src/client/components/share.stories.ts

```diff
@@ -16,16 +17,23 @@ interface ShareArgs {
     prompt?: string,
 }
 
-const Template = (args: ShareArgs) => html`
-    <dwac-share .target="${args.target ? new URL(args.target) : undefined}"
-            .articleTitle="${args.articleTitle}"
-            .prompt="${args.prompt}">
-    </dwac-share>
-`;
-
-export const Primary = Template.bind({});
-(Primary as any).args = {
-    target: 'https://blog.dwac.dev/',
-    articleTitle: 'Devel without a Cause',
-    prompt: 'Check out my blog!',
+export const primary = ({
+    target = 'https://blog.dwac.dev/',
+    articleTitle = 'Devel without a Cause',
+    prompt = 'Check out my blog!',
+}: ShareArgs = {}) => {
```

Comment:
I see, so its not that they need to be declared here, its that they need a default value to be compatible with Storybook. I guess it makes sense that a component may have some required properties and it would be up to the story to provide a default value.

Ideally, I would love this to be a documentable feature which could be displayed in Storybook. Some kind of `@required` JSDoc tag which flags the knob in Storybook and displays in the docs. I guess I'm weird in my philosophy on required properties in web components. I can also see some value in specifying a default in the story just so it populates some valid data for a user. Otherwise a required knob might be confusing to a user who doesn't understand what to put in it.